### ISSUE TITLE
Highlighting error "Less than 2 subSpans.size()" due to LUCENE-7231

### ISSUE THREAD

**Issue by thomas11** [elastic/elasticsearch#19844]
**Elasticsearch version**: 2.3.3

**Plugins installed**: [analysis-kuromoji, analysis-phonetic, analysis-smartcn, analysis-stempel]

**JVM version**: 1.8.0_u92

For some search requests with highlighting, we run into https://issues.apache.org/jira/browse/LUCENE-7231. The Lucene issue has a repro.

This bug has been fixed in Lucene 5.5.2. Any chance of Elasticsearch upgrading in the next minor release?

Thanks

**Comment by jasontedor** [comment]
Closed by #19496. Lucene 5.5.2 will be included in the forthcoming 2.4.0 release of Elasticsearch.

**Comment by thomas11** [comment]
Thanks for the prompt response @jasontedor! Since this bug breaks functionality on our side I'd like to know when to expect 2.4.0, roughly. Any idea?

**Comment by naag** [comment]
This still happens on release 5.0.0-alpha5:

```
{
name: "ekJiZIT",
cluster_name: "elasticsearch",
version: {
number: "5.0.0-alpha5",
build_hash: "d327dd4",
build_date: "2016-08-04T08:59:39.568Z",
build_snapshot: false,
lucene_version: "6.1.0"
},
tagline: "You Know, for Search"
}
```

JVM version is: 

```
# java -version
openjdk version "1.8.0_102"
OpenJDK Runtime Environment (build 1.8.0_102-8u102-b14.1-1~bpo8+1-b14)
OpenJDK 64-Bit Server VM (build 25.102-b14, mixed mode)
```

Does ES 5.0.0-alpha5 already include the Lucene fix? Thanks!

**Comment by jasontedor** [comment]
> Does ES 5.0.0-alpha5 already include the Lucene fix?

Yes.

**Comment by thomas11** [comment]
There is a related bug that results in the same error when using
MultiPhraseQuery. It's fixed in the latest Lucene 6.2.1, but not in 6.1.

https://issues.apache.org/jira/plugins/servlet/mobile#issue/LUCENE-7417
